### PR TITLE
Support account lock/unlock in creating/altering user

### PR DESCRIPTION
**Pull request by CbcWestwolf**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/37052

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: sql-statements/sql-statement-create-user.md [hunk 1]
```diff
@@ -29,6 +29,8 @@ AuthOption ::=
 StringName ::=
     stringLit
 |   Identifier
+
+LockOption ::= ( 'ACCOUNT' 'LOCK' | 'ACCOUNT' 'UNLOCK' )?
 ```
 
 ## 示例
```
Comment (ran-huang) [inline sql-statements/sql-statement-create-user.md:36]: Can you provide an example using the `ACCOUNT LOCK` clause?

Comment (CbcWestwolf) [reply sql-statements/sql-statement-create-user.md:36]: Updated. PTAL.

File: sql-statements/sql-statement-alter-user.md [hunk 2]
```diff
@@ -81,6 +81,20 @@ SHOW CREATE USER 'newuser';
...
+ALTER USER 'newuser' ACCOUNT LOCK;
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
+
+> **注意：**
+>
+> 不要使用 `ACCOUNT UNLOCK` 解锁一个[角色（`Role`）](sql-statement-create-role.md)，否则通过该角色可以免密码登入 TiDB。
```
Comment (ran-huang) [inline sql-statements/sql-statement-alter-user.md:96]:
```suggestion
> 不要使用 `ACCOUNT UNLOCK` 解锁一个[角色 (`Role`)](sql-statement-create-role.md)，否则通过被解锁的角色可以免密码登入 TiDB。
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

File: sql-statements/sql-statement-alter-user.md [hunk 3]
```diff
@@ -79,6 +81,20 @@ SHOW CREATE USER 'newuser';
...
+ALTER USER 'newuser' ACCOUNT LOCK;
+```
+
+```
+Query OK, 0 rows affected (0.02 sec)
+```
+
+> **注意：**
+>
+> 不要使用 `ACCOUNT UNLOCK` 解锁一个[角色 (`Role`)](/sql-statements/sql-statement-create-role.md)，否则通过被解锁的角色可以免密码登入 TiDB。
```
Comment (Oreoxmt) [inline sql-statements/sql-statement-alter-user.md:96]:
```suggestion
> 不要使用 `ACCOUNT UNLOCK` 解锁一个[角色 (Role)](/sql-statements/sql-statement-create-role.md)，否则通过被解锁的角色可以免密码登入 TiDB。
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by ljun0712 — APPROVED** [review]

**Pull request merged**